syntax = "proto3";

import "map_renderer.proto";
import "transport_router.proto";

package transport_catalogue_proto;

message Coordinates {
    double latitude = 1;
    double longitude = 2;
}

message DistanceBetweenStops {
    string from_stop = 1;
    string to_stop = 2;
    uint32 distance = 3;
}

message Stop {
    string name = 1;
    Coordinates coordinates = 2;
    uint32 id = 3;
}

message Bus {
    string name = 1;
    repeated string name_stops = 2;
    bool is_ring = 3;
}

message TransportCatalogue {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated DistanceBetweenStops distances = 3;
    map_renderer_proto.MapVisualizationSettings render_settings = 4;
    transport_router_proto.RouterSettings router_settings = 5;
}